packages/app/src/cli/models/extensions/extension-instance.ts

```diff
@@ -139,6 +142,8 @@ export class ExtensionInstance<TConfiguration extends BaseConfigType = BaseConfi
       const config = this.configuration as unknown as FunctionConfigType
       this.outputPath = joinPath(this.directory, config.build.path ?? joinPath('dist', 'index.wasm'))
     }
+
+    this.appConfiguration = options.appConfiguration
```

Comment:
It's a bit weird that each extension has a copy of the app configuration. Not sure how compatible this would be with `dev` where the app config can be updated, but it won't update every extension's copy 🤔

Reply:
Hmm. I suppose an alternate would be to pass in the latest config when the deploy function runs? Or, reload the app when the configuration is changed at runtime -- I suppose this latter is more correct, there are other bugs you could experience if you have a modified configuration object but the appinterface is now out of date

Reply:
Ok, will go for first option. I think we should think about how to wrap more of this access and mutation up in the app though. Like, we shouldn't be working with an isolated extension, or taking the config out of the app and changing it